src/test/java/com/work/javaconsumer/service/PlayerConsumerServiceTest.java

```diff
@@ -25,25 +25,42 @@ class PlayerConsumerServiceTest {
     PlayerConsumerService playerConsumerService;
 
     @Test
-    public void should_save_one_player() {
+    public void should_find_one_player()  {
         //given
-        Player newPlayer = new Player("Bob", "Bobzoor@gmail.com", 0);
-        RegisterPlayerDTO dto = new RegisterPlayerDTO("Bob", "Bobzoor@gmail.com", 0);
+        RegisterPlayerDTO dto = new RegisterPlayerDTO("Bobba", "Bobzoor@gmail.com", 0);
+        Player newPlayer = new Player(dto.getName(), dto.getEmail(), dto.getBalance());
+
+        when(playerRepository.findByName(newPlayer.getName())).thenReturn(Optional.of(newPlayer));
+
+        //when
+        playerConsumerService.registerPlayer(dto);
+
+        //then
+        Assertions.assertEquals(dto.getName(), newPlayer.getName());
+        verify(playerRepository, times(1)).findByName(newPlayer.getName());
+    }
+
+    @Test
+    public void should_save_one_player()  {
+        //given
+        RegisterPlayerDTO dto = new RegisterPlayerDTO("Bobba", "Bobzoor@gmail.com", 0);
+        Player newPlayer = new Player(dto.getName(), dto.getEmail(), dto.getBalance());
 
         when(playerRepository.save(any(Player.class))).thenReturn(newPlayer);
 
         //when
         playerConsumerService.registerPlayer(dto);
 
         //then
+        Assertions.assertEquals(dto.getName(), newPlayer.getName());
```

Comment:
For suggestion testing
```suggestion
        Assertions.assertEquals(newPlayer.getName(), dto.getName());
```